### PR TITLE
cleanup `computed.teardown`

### PR DESCRIPTION
**Pull request by bekzod**

### CODE REVIEW

File: packages/ember-metal/lib/computed.js [hunk 1]
```diff
@@ -426,11 +426,11 @@ ComputedPropertyPrototype._set = function computedPropertySet(obj, keyName, valu
 };
 
 /* called before property is overridden */
-ComputedPropertyPrototype.teardown = function(obj, keyName) {
+ComputedPropertyPrototype.teardown = function(obj, keyName, _meta) {
   if (this._volatile) {
     return;
   }
-  let meta = metaFor(obj);
```
Comment (bekzod) [inline packages/ember-metal/lib/computed.js]: it is pure speculation, I think it should be `peekMeta`

Comment (stefanpenner) [inline packages/ember-metal/lib/computed.js]: yup this is correct, but we will now need to guard against `meta === null`.

something like
```js
+  let meta = _meta || peekMeta(obj);
 
if ( meta === null) { return; }
```

**Review by stefanpenner — CHANGES_REQUESTED** [review]

File: packages/ember-metal/lib/computed.js (hunk 1 above)
Comment (stefanpenner) [inline packages/ember-metal/lib/computed.js]: another question though is, if teardown isn't public and the only place we call it we already provide meta, then maybe we don't need to do a meta lookup at all? What are your thoughts?

Comment (bekzod) [inline packages/ember-metal/lib/computed.js]: sure...doing that

File: packages/ember-metal/lib/computed.js [hunk 2]
```diff
@@ -426,13 +426,12 @@ ComputedPropertyPrototype._set = function computedPropertySet(obj, keyName, valu
 };
 
 /* called before property is overridden */
-ComputedPropertyPrototype.teardown = function(obj, keyName) {
-  if (this._volatile) {
+ComputedPropertyPrototype.teardown = function(obj, keyName, meta) {
+  if (this._volatile || !meta) {
```
Comment (stefanpenner) [inline packages/ember-metal/lib/computed.js]: i believe it may be impossible for `meta` to be null here. Or..?

**Review by stefanpenner — CHANGES_REQUESTED** [review]

File: packages/ember-metal/lib/computed.js (hunk 2 above)
Comment (bekzod) [inline packages/ember-metal/lib/computed.js]: I believe too ... so no guard ?

**Review by stefanpenner — APPROVED** [review]

**Comment by stefanpenner** [comment]
restarting sauce

**Comment by locks** [comment]
image

**Pull request merged**